docs/src/structuring_element.md

```diff
@@ -0,0 +1,197 @@
+neighborhood. This page explains this structuring element concept, and how ImageMorphology supports
+the general SEs without compromising the performance on the most commonly used special SE cases.
+
+## The erosion example
+
+The erosion `erode` function in its simplest 1-dimensional case can be defined as
+
+$$\varepsilon_A[p] = min(A[p-1], A[p], A[p+1])$$
+
+Because the output value at position $p$ not only depends on its own pixel `A[p]` but also on
+its neighborhood values `A[p-1]` and `A[p+1]`, we call this type of operation a _neighborhood
+image transformation_.
+
+Now comes the question: **if we try to generalize the `erode` function, what should we do?** --
+we would like to generalize the concept of "neighborhood".
+
+## Two neighborhood representations
+
+By saying "$\Omega_p$ is the neighborhodd of $p$", we are expressing `p in Ωₚ` in plain Julia. For
+performance consideration, this `Ωₚ` is usually generated from the `(p, Ω)` pair; we call this
+template `Ω` a _structuring element_. There are two ways to expressing this `Ω`:
+
+- **displacement offset**: a list of `CartesianIndex` to inidcate the offset to the center point `p`
+- **connectivity mask**: a bool array mask to indicate the connectivity to the center point `p`
+
+For instance, in the following code block we build a commonly named C4 connectivity in the 2-dimensional case:
+
+```@example concept_se
+# displacement offset
+Ω_offsets = [
+    CartesianIndex(-1, 0),
+    CartesianIndex(0, -1),
+    CartesianIndex(0, 1),
+    CartesianIndex(1, 0),
+]
+
+# connectivity mask
+Ω_bool = Bool[
+    0 1 0
+    1 1 1
+    0 1 0
+]
+nothing #hide
+```
+
+If `p=(3, 3)`, then we know `p=(3, 4)` is in `Ωₚ`.
+
+Now, back to the erosion example. Based on the displacement offset representation, the simplest
+generic version of `erode` can be implemented quite simply:
+
+```@example concept_se
+# Illustration purpose only, do not use this in your real-world project
+function my_erode(A, Ω)
+    out = similar(A)
+    R = CartesianIndices(A)
+    for p in R
+        Ωₚ = filter!(q->in(q, R), Ref(p) .+ Ω)
+        # here we don't assume p in Ωₚ
+        out[p] = min(A[p], minimum(A[Ωₚ]))
+    end
+    return out
+end
+nothing #hide
+```
+
+```@example concept_se
+using ImageMorphology
+using ImageBase
+using TestImages
+
+img = Gray.(testimage("morphology_test_512"))
+img = Gray.(img .< 0.8)
+img_e = my_erode(img, Ω_offsets)
+mosaic(img, img_e; nrow=1)
+```
+
+As you may realize, the displacement offset representation is convenient to use when implementing
+algorithms, but it is hard to visualize. In contrast, the connectivity mask is not so convenient to
+use when implementing algorithms, but it is easy to visualize. For instance, one can very easily
+understand the following SE at the first glance:
+
+```@example concept_se
+Ω = Bool[1 1 1; 1 1 0; 0 0 0] # hide
+```
+
+but not
+
+```@example concept_se
+strel(CartesianIndex, Ω) # hide
```

Comment:
`strel` hasn't been mentioned yet, so it's not clear whether this should be easy or difficult to comprehend.

Reply:
There is a carefully used `# hide` mark so that only the output is shown here:

<img src="https://user-images.githubusercontent.com/8684355/167310098-d3e3e207-3488-4e9a-b1d9-f5e035b8d292.png" width=512>